### PR TITLE
Move to the current API of qiskit-nature

### LINKED ISSUE #87: qiskit-cold-atom not compatible with qiskit-nature 0.7.1
### Informations

- **Qiskit Cold atoms version**:

current branch

- **Python version**:

3.10

- **Operating system**:

all

### What is the current behavior?

Trying to run tests and install qiskit-cold-atom requires to fix `qiskit-nature` to <= 0.6.2. Otherwise it raises several errors in linting and testing. The main problem seems to be:

 No name 'operators' in module 'qiskit_nature'

### Steps to reproduce the problem

All problems are produced in [this test run](https://github.com/qiskit-community/qiskit-cold-atom/actions/runs/7157540648/job/19488532688?pr=86).

### What is the expected behavior?

We should be able to cleanly work with qiskit-nature in its most current version.

### PR DESCRIPTION
**Pull request by fretchen**
### Summary

This attempts to fix #87 and most likely also #88

### Details and comments

However, we have to jump from the syntax of 0.4 to the syntax of 0.7 for `qiskit-nature` and this seems to be a fairly major shift. I think that I might have found most of the operators, but some problems and questions remain:
I think that I might have found most of the operators, but some problems and questions remain:

- It would seem that `SparseLabelOp` is the new name of `SecondQuantizedOp` ? Not sure what has to be changed to make it work.
- The `register_length` seems to not exist anymore and had to be updated to `num_spin_orbitals` or `num_spins`.
- For the moment the tests still fail with `'Gate rlx has no defined generator'`. No idea why.
- I think that we should also enable dependabot updates to avoid another massive version jump like this one in the future.

### CODE REVIEW

File: releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
+  - |
+    Upgrade the syntax to be compatible with the version 0.7 of the `qiskit-nature`.
```
Comment (eggerdj) [inline releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml:4]: Can you add a few more lines to describe the changes? Also is there anything that directly impacts the users of `qiskit-cold-atom`? That might also be good to describe.

File: requirements.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 numpy>=1.17
 scipy>=1.4
 requests>=2.25.1
-qiskit-terra>= 0.17.1, <=0.25.2.1
-qiskit-nature>=0.4.4,<=0.6.2
-qiskit-aer>=0.8.0, <=0.12.2
+qiskit-terra>= 0.45.1
+qiskit-nature>=0.7.1
+qiskit-aer>=0.8.0
```
Comment (eggerdj) [inline requirements.txt:6]: 💯

**Review by eggerdj — CHANGES_REQUESTED** [review]
This looks really good. Thanks a lot for all the effort. From my perspective there are still two open points:
* A bit more description in the reno
* Resolve the merge conflicts

File: releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml (hunk 1 above)
Comment (fretchen) [reply releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml:4]: So for the users I do not think that there is anything changing except that you should now use the syntax of `qiskit-nature` 0.7. @eggerdj maybe you could propose what we mainly changed there as those come mostly from you ?

File: qiskit_cold_atom/fermions/fermion_gate_library.py [hunk 3]
```diff
@@ -172,38 +172,26 @@ def operator_to_mat(
                 sign = 1
 
                 # in reverse, loop over all individual fermionic creators/annihilators in the opstring:
-                for k, symbol in reversed(list(enumerate(opstring))):
+                for k, symbol in reversed(list(enumerate(opstrings))):
                     if symbol == "I":
                         continue
```
Comment (LaurinFischer) [inline qiskit_cold_atom/fermions/fermion_gate_library.py:177]: These three lines should be replaced with 
```suggestion
                for symbol, k in reversed(opstrings):
```
It seems that when resolving the merge conflict you selected the version from the current main, whereas this new version is the one we need to be compatible with qiskit-nature.

File: releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml (hunk 1 above)
Comment (LaurinFischer) [inline releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml:4]: Let's put this:

```suggestion
  - |
    Upgrade the syntax to be compatible with the version 0.7 of the `qiskit-nature`.
    This affects the definitions of FermionicOp as the generator of a FermionicGate and SpinOp for a SpinGate.
    FermionicOp now are now built from sparse labels with creators ("+") and annihilators ("-") and SpinOp allows for sparse labels with "X", "Y" and "Z". 
    Please see the documentation of qiskit-nature for more details. 
```

**Review by LaurinFischer — CHANGES_REQUESTED** [review]
With these changes the tests should be good and the releasenote a bit more detailed.

**Comment by fretchen** [comment]
@eggerdj I think that the comments should now be taken into account and we are again ready for review.

File: releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml [hunk 4]
```diff
@@ -2,4 +2,7 @@
 upgrade:
   - |
     Upgrade the syntax to be compatible with the version 0.7 of the `qiskit-nature`.
+    This affects the definitions of FermionicOp as the generator of a FermionicGate and SpinOp for a SpinGate.
+    FermionicOp now are now built from sparse labels with creators ("+") and annihilators ("-") and SpinOp allows for sparse labels with "X", "Y" and "Z".
```
Comment (eggerdj) [inline releasenotes/notes/upgrade-of-qiskit-nature-f808e73a446c1c93.yaml:6]:
```suggestion
    FermionicOp are now built from sparse labels with creators ("+") and annihilators ("-") and SpinOp allows for sparse labels with "X", "Y" and "Z". 
```

**Review by eggerdj — APPROVED** [review]
Looks good to me. Thanks a lot for this!

**Review by LaurinFischer — APPROVED** [review]

**Pull request merged**